client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteSchemasAdminImpl.java

```diff
@@ -0,0 +1,234 @@
+import org.infinispan.client.hotrod.RemoteSchemasAdmin;
+import org.infinispan.client.hotrod.impl.operations.ManagerOperationsFactory;
+import org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher;
+import org.infinispan.client.hotrod.logging.Log;
+import org.infinispan.client.hotrod.logging.LogFactory;
+import org.infinispan.commons.internal.InternalCacheNames;
+import org.infinispan.commons.util.concurrent.CompletionStages;
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.schema.Schema;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.infinispan.client.hotrod.Flag.FORCE_RETURN_VALUE;
+import static org.infinispan.client.hotrod.impl.RemoteCacheManagerAdminImpl.string;
+
+public class RemoteSchemasAdminImpl implements RemoteSchemasAdmin {
+    public static final Log log = LogFactory.getLog(RemoteSchemasAdminImpl.class, Log.class);
+
+    private final ManagerOperationsFactory operationsFactory;
+    private final OperationDispatcher operationDispatcher;
+    private final RemoteCache<String, String> protostreamCache;
+    private static final byte[] CREATE = string("c");
+    private static final byte[] UPDATE = string("u");
+    private static final byte[] SAVE = string("s");
+    private static final byte[] FORCE = string("f");
+
+    protected RemoteSchemasAdminImpl(ManagerOperationsFactory operationsFactory, OperationDispatcher operationDispatcher, RemoteCacheManager cacheManager) {
+        this.operationsFactory = operationsFactory;
+        this.operationDispatcher = operationDispatcher;
+        this.protostreamCache = cacheManager.<String, String>getCache(InternalCacheNames.PROTOBUF_METADATA_CACHE_NAME).withFlags(FORCE_RETURN_VALUE);
+    }
+
+   @Override
+   public CompletionStage<Boolean> existsAsync(String schemaName) {
+      return protostreamCache.containsKeyAsync(schemaName);
+   }
+
+   @Override
+   public CompletionStage<Optional<Schema>> getAsync(String schemaName) {
+      return protostreamCache.getWithMetadataAsync(schemaName)
+            .thenApply(e -> getSchemaContent(schemaName, e))
+            .exceptionally(ex -> {
+               log.crudSchemaError(schemaName, ex);
+               throw new InfinispanException(ex.getMessage(), ex);
+            });
+   }
+
+    @Override
+    public CompletionStage<SchemaOpResult> createAsync(Schema schema) {
+       return callCreateOrUpdateSchema(schema, CREATE, false);
+    }
+
+   @Override
+   public CompletionStage<Map<String, SchemaOpResult>> createAsync(FileDescriptorSource fileDescriptorSource) {
+      return createOrUpdateFileDescriptors(fileDescriptorSource, schema -> createAsync(schema));
```

Comment:
`return createOrUpdateFileDescriptors(fileDescriptorSource, this::createAsync);` and below